### DISCUSSION TITLE
Relative path to featured image in page bundle not working

### DISCUSSION THREAD

**Discussion by metbril** [Lednerb/bilberry-hugo-theme#422] [Q&A]
I use page bundles for my articles. The featured image is applied through front matter. The theme currently supports this with the `featuredImage` front matter parameter.

When adding the front matter, my featured image is not shown as expected.

Both `featuredImage: some-image.jpg` and `featuredImage: ./some-image.jpg` are rendered as `<img src="/some-image.jpg" alt="">`.

My expectation is, that the path is rendered as specified, so `<img src="some-image.jpg" alt="">` and `<img src="./some-image.jpg" alt="">` respectively.

**Comment by igor-baiborodine** [comment]
@metbril, could you please confirm that the main issue here is that the featured image, defined via the front matter variable `featuredImage` with a relative path in a page bundle, is not shown **at all**?

**Comment by metbril** [comment]
I can confirm it is not shown at all. 

The <img> tag currently refers to the root of the site, but should refer to `/path/to/page/bundle/some-image.jpg` or a relative address.

front matter:

```md
---
title: "Genieten van De Nes"
slug: "genieten-van-de-nes"
date: 2022-02-13T22:29:00+01:00
author: Robert van Bregt
tags: [buiten,wandelen,route,track]
categories:
  - Buiten
featuredImage: "nespolder-minibrug.jpg"
---
```

rendered page:

```html
    <div class="featured-image">
        <a href="http://localhost:1313/posts/2022/02/genieten-van-de-nes/">
            <img src="/nespolder-minibrug.jpg" alt="">
        </a>
    </div>
```

screenshot:

<img width="853" alt="Schermafbeelding 2022-07-29 om 14 14 42" src="https://user-images.githubusercontent.com/559346/181756314-4fe4a48d-9c3d-4110-94ba-36349207c255.png">

page bundle:

<img width="249" alt="Schermafbeelding 2022-07-29 om 14 16 24" src="https://user-images.githubusercontent.com/559346/181756533-5f8bbc31-b457-49a1-b485-508671329a98.png">

**Comment by igor-baiborodine** [comment]
@metbril, in README, I added a new section named [Featured Image](https://github.com/Lednerb/bilberry-hugo-theme#featured-image):

![image](https://user-images.githubusercontent.com/1027701/181825178-e936a49c-3ca0-4472-9328-5e1016901c10.png)

In your case, the image path is not relative to the base URL. That's why it is not working. When using the page bundle, you can use the `featuredImage` front matter variable, but in this case, the featured image will not be cropped and resized.

Please see these test articles I created back then on the Bilberry Sandbox:
* Article 1
  https://github.com/igor-baiborodine/bilberry-hugo-theme-sandbox/tree/master/content/article/test-featured-image-resizing
  https://www.bilberry-sandbox.kiroule.com/article/test-featured-image-resizing/

* Article 2
  https://github.com/igor-baiborodine/bilberry-hugo-theme-sandbox/tree/master/content/article/test-featured-image-resizing-via-front-matter
  https://www.bilberry-sandbox.kiroule.com/article/test-featured-image-resizing-via-front-matter/

**Comment by metbril** [comment]
Too bad you have closed this issue. I still believe that the current way it works for page bundles is not how it should.

Page bundles are intended to be self contained. So relative addressing should be possible. This is how most other themes work.

In bilberry when you currently change the permalink structure, for example, absolute paths would need to be changed in the front matter.

The workaround should be to just copy any featured image to one with the hard coded filename, but then I would have 2 copies of that image in my file bundle when I need that somewhere else in the post with a regular filename.

In the current `featured-image.html` partial, only one line should need to be changed from

```
            <img src="{{ .Params.featuredImage | relURL}}" alt="">
```

to

```
            <img src="{{ .Params.featuredImage }}" alt="">
```

to make it work.

What is the purpose of the `relURL`?

**Reply by andreagrandi** [reply]
I'm having the same issue and honestly I'd rather change the template (thanks for suggesting how) than my content. I'm trying to migrate from Pelican to Hugo and I'm finding it very unusual (and not good practice, imho) that each theme has to re-define its own way of configuring very common stuff (like social links, footer copyright information, open graph metadata etc....)

**Reply by igor-baiborodine** [reply]
@andreagrandi, here is how you can change the template for the featured image:

1. Copy the `v4/layouts/partials/featured-image.html` file from the Bilberry Hugo theme to `<your-website-root>/layouts/partials/featured-image.html`: 

![image](https://github.com/Lednerb/bilberry-hugo-theme/assets/1027701/7763921f-4861-4b41-9348-e02a866321c7)

2. Modify the copied template as needed:

![image](https://github.com/Lednerb/bilberry-hugo-theme/assets/1027701/5fe6cfa2-909b-4f54-9645-951d292ab883)

**Comment by igor-baiborodine** [comment]
@metbril, you can read more on the `relUrl` function [here](https://gohugo.io/functions/relurl/).

As for the relative URLs in your particular case, you can set the `relativeUrls` property to `true` in your configuration file to rewrite all relative URLs to be relative to the current content. Here is the Hugo [documentation](https://gohugo.io/functions/relurl/) on this property. 

I made a quick test in my local dev with the Bilberry Sandbox:
![image](https://user-images.githubusercontent.com/1027701/182046466-3c2b5813-b8b5-434c-824f-15e202f831d1.png)

![image](https://user-images.githubusercontent.com/1027701/182046498-508001f6-d53b-4de2-96aa-3ee9e81670f4.png)

![image](https://user-images.githubusercontent.com/1027701/182046519-927c9d84-daba-422c-91c9-0a39316fb3f8.png)

I'm moving this issue to Discussions since it can be helpful for other users.

**Comment by metbril** [comment]
It's been some time since, but I'm still going to try the `relativeUrls` parameter. Thanks so far.
